src/renderer/services/neo.js

```diff
@@ -573,6 +704,98 @@ export default {
+          canPull, isETHToken, abi, availableSupply) => {
+          const systemAssetHolding = {
+            assetId,
+            balance: new BigNumber(balance),
+            symbol,
+            name,
+            contractBalance: new BigNumber(0),
+            totalBalance: new BigNumber(0),
+            decimals,
+            isUserAsset: true,
+            canPull,
+            isETHToken,
+            abi,
+            availableSupply,
+          };
+          holdings.push(systemAssetHolding);
+        };
+
+        // ETH System Holding
+
+        const ethAddress = currentWallet.ethAddress;
+        let balEthTemp;
+        let balEth;
+
+        try {
+          balEthTemp = await web3.eth.getBalance(ethAddress);
+          balEth = toBigNumber(balEthTemp).dividedBy(10 ** 18);
+        } catch (e) {
+          balEthTemp = 0;
+          balEth = 0;
+          alerts.networkException(e);
+        }
+
+        pushNewHolding('0x0000000000000000000000000000000000000000',
+          'Ethereum', 'ETH', balEth, 18, undefined, undefined, undefined, undefined);
+
+        // BTC System Holding
+
+        let balBTCtemp;
+        let balBTC;
+        const walletClient = currentWallet.btcWalletClient;
+
+        try {
+          balBTCtemp = await bwclient.getBTCBalanceByAddress(walletClient);
+          balBTC = balBTCtemp.availableAmount;
+        } catch (e) {
+          balBTCtemp = 0;
+          balBTC = 0;
+          alerts.networkException(e);
+        }
+
+        const addressesTemp = balBTCtemp.byAddress;
+        const addressArr = [];
+
+        addressesTemp.forEach((addressData) => {
+          addressArr.push(addressData.address);
+        });
+
+        this.generateNewBTCAddress(addressArr, currentWallet, currentNetwork, walletClient);
+
+        const balBTCset = toBigNumber(balBTC).dividedBy(10 ** 8);
+        pushNewHolding('1111111111111111111111111111111111',
+          'Bitcoin', 'BTC', balBTCset, 8, undefined, undefined, undefined, undefined);
+
+        // ERC20 token balance
+
+        _.values(userAssets).forEach(async (asset) => {
+          if (asset.isETHBased) {
+            pushNewHolding(asset.assetId, asset.name, asset.symbol,
+              0, asset.decimals, true, true, asset.abi, 0);
+          }
+        });
+
+        _.values(holdings).forEach(async (holding) => {
+          if (holding.isETHToken) {
+            try {
+              const tokenContract = new web3.eth.Contract(JSON.parse(holding.abi), holding.assetId);
+              const balanceTokenTemp = await tokenContract.methods.balanceOf(ethAddress).call();
+              holding.balance = toBigNumber(balanceTokenTemp).dividedBy(10 ** (holding.decimals));
+
+              if (network.getSelectedNetwork().net === 'TestNet') {
+                holding.availableSupply = await tokenContract.methods.totalSupply().call() / (10 ** holding.decimals);
+              }
+            } catch (e) {
+              holding.balance = 0;
```

Comment:
Same as above; it's probably better to show the previous balance in the presence of a network error than to make the balance show as 0.